cxx-squid/src/main/java/org/sonar/cxx/preprocessor/ExpressionEvaluator.java

```diff
@@ -144,7 +145,8 @@ private BigInteger evalComplexAst(AstNode exprAst) {
     } else if ("functionlikeMacro".equals(nodeType)) {
       return evalFunctionlikeMacro(exprAst);
     } else {
-      throw new EvaluationException("Unknown expression type '" + nodeType + "'");
+      LOG.error("'evalComplexAst' Unknown expression type '" + nodeType + "' for AstExt '" + exprAst.getToken() + "', assuming 0");
+      return BigInteger.ZERO;
```

Comment:
this one i am not sure, but taking the example of the others that return Zero i supose if node type not found then we retrun zero, this also prevent leaking stacktraces upstreams